sklearn/svm/_classes.py

```diff
@@ -37,9 +54,14 @@ class LinearSVC(LinearClassifierMixin, SparseCoefMixin, BaseEstimator):
         square of the hinge loss. The combination of ``penalty='l1'``
         and ``loss='hinge'`` is not supported.
 
-    dual : bool, default=True
+    dual : "auto" or bool, default=True
         Select the algorithm to either solve the dual or primal
         optimization problem. Prefer dual=False when n_samples > n_features.
+        `dual="auto"` will choose the best option considering `n_samples`,
+        `n_features`, `loss` and `penalty`.
```

Comment:
I'm not sure if we can say _best_ here. I rather say maybe fastest? Or simply saying, automatically choosing, based on the input, and explain how it's done in the user guide with a link to user guide, or have it all here.